File: Java/ex3.java
class ContaBancaria {
    private String titular;
    private double saldo; 

    public ContaBancaria(String titular, double saldoInicial) {
        this.titular = titular;
        this.saldo = saldoInicial; 
    }

  
    public void depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            System.out.printf("Depósito de R$%.2f realizado com sucesso. Saldo atual: R$%.2f%n", valor, saldo);
        } else {
            System.out.println("Valor de depósito deve ser positivo.");
        }
    }

  
    public void sacar(double valor) {
        if (valor > saldo) {
            System.out.println("Saque não realizado. Saldo insuficiente.");
        } else {
            saldo -= valor;
            System.out.printf("Saque de R$%.2f realizado com sucesso. Saldo atual: R$%.2f%n", valor, saldo);
        }
    }

    public double getSaldo() {
        return saldo;
    }

    
    public void exibirDetalhes() {
        System.out.println("Titular: " + titular + ", Saldo: R$" + saldo);
    }

    public static void main(String[] args) {
        
        ContaBancaria conta = new ContaBancaria("Maria", 500.00);

        
        conta.exibirDetalhes();

        
        conta.depositar(150);
        conta.sacar(100);
        conta.sacar(600); 
        conta.exibirDetalhes();
    }
}

File: python/ex3.py


class BankAccount:
    def __init__(self, titular, saldo_inicial=0):
        self.__saldo = saldo_inicial  
        self.titular = titular

    #
    def deposit(self, valor):
        if valor > 0:
            self.__saldo += valor
            print(f"Depósito de R${valor:.2f} realizado com sucesso.")
        else:
            print("O valor de depósito deve ser positivo.")

    
    def withdraw(self, valor):
        if valor > 0:
            if valor <= self.__saldo:
                self.__saldo -= valor
                print(f"Saque de R${valor:.2f} realizado com sucesso.")
            else:
                print("Saldo insuficiente.")
        else:
            print("O valor de saque deve ser positivo.")

    
    def display_balance(self):
        return f"Saldo atual: R${self.__saldo:.2f}"


def main():
    conta = BankAccount("Leonardo Sobrinho", 1000)
    
    
    print(conta.display_balance())
    
    
    conta.deposit(500)
    print(conta.display_balance())
    
    conta.withdraw(300)
    print(conta.display_balance())
    
    conta.withdraw(1500)  


if __name__ == "__main__":
    main()
